internal/clients/influxdb/influxdb.go

```diff
@@ -21,32 +21,30 @@ type Config struct {
 	Username           string        `env:"ADMIN_USER"            envDefault:"mainflux"`
 	Password           string        `env:"ADMIN_PASSWORD"        envDefault:"mainflux"`
 	DbName             string        `env:"DB"                    envDefault:"mainflux"`
+	Bucket             string        `env:"BUCKET"                envDefault:"mainflux-bucket"`
+	Org                string        `env:"ORG"                   envDefault:"mainflux"`
+	Token              string        `env:"TOKEN"                 envDefault:"mainflux-token"`
+	DBUrl              string        `env:"DBURL"                   envDefault:""`
 	UserAgent          string        `env:"USER_AGENT"            envDefault:"InfluxDBClient"`
 	Timeout            time.Duration `env:"TIMEOUT"` // Influxdb client configuration by default there is no timeout duration , this field will not have fallback default timeout duration Reference: https://pkg.go.dev/github.com/influxdata/influxdb@v1.10.0/client/v2#HTTPConfig
 	InsecureSkipVerify bool          `env:"INSECURE_SKIP_VERIFY"  envDefault:"false"`
 }
 
 // Setup load configuration from environment variable, create InfluxDB client and connect to InfluxDB server
-func Setup(envPrefix string) (client.HTTPClient, error) {
+func Setup(envPrefix string, ctx context.Context) (influxdb2.Client, error) {
 	config := Config{}
 	if err := env.Parse(&config, env.Options{Prefix: envPrefix}); err != nil {
 		return nil, errors.Wrap(errConfig, err)
 	}
-	return Connect(config)
+	return Connect(config, ctx)
 }
 
 // Connect create InfluxDB client and connect to InfluxDB server
-func Connect(config Config) (client.HTTPClient, error) {
-	address := fmt.Sprintf("%s://%s:%s", config.Protocol, config.Host, config.Port)
-	clientConfig := client.HTTPConfig{
-		Addr:      address,
-		Username:  config.Username,
-		Password:  config.Password,
-		UserAgent: config.UserAgent,
-		Timeout:   config.Timeout,
-	}
-	client, err := client.NewHTTPClient(clientConfig)
-	if err != nil {
+func Connect(config Config, ctx context.Context) (influxdb2.Client, error) {
+	client := influxdb2.NewClient(config.DBUrl, config.Token)
+	ctx, cancel := context.WithTimeout(ctx, config.Timeout)
```

Comment:
Looks like this timeout is not configured correctly (or does not have the correct default value), so `Ready` always fails due to expired context.